content/Phoenix for Rails developers - Bruno Vezoli.md

```diff
@@ -0,0 +1,347 @@
+
+Since that time I’ve tried many other web development frameworks in a wide variety of languages but I never felt at home with any of them. Even if many tried to replicate what Rails did there was always missing, whether it was because of the language or some decision where it diverged they always felt like mere copies to me.
+
+### Phoenix
+
+Enter Phoenix, a web framework built on top of Elixir, a functional language with a Ruby-like syntax that runs on top of the Erlang virtual machine. This framework took after many of the features that make Rails what it is: MVC architecture, an opinionated file structure, included tools such as an ORM, a template engine and a routing layer. But it never tried to be another copy, instead it diverged with time and took many decisions that fit better the functional paradigm and the different language, all without sacrificing productivity or developer happiness.
+
+In time, this led to an independent web framework but with that special flavor we all love from Rails. There’s one difference though, since Elixir was built on top of Erlang, which was designed to build massively scalable applications, all apps built with Phoenix inherit this perk. This means that you can code away without having to worry how your app will perform once you reach the big leagues. This is something with which many Rails apps struggle and it’s the main reason why Jose Valim developed Elixir in the first place.
+
+But enough preaching, let’s see some code!
+
+
+### Comparison by example
+
+This post does not intend to be a complete guide on Phoenix, it’s a big framework with lots of things to get into. Instead, I’ll show a side by side comparison of how things are done in both frameworks using the scaffolding they provide to throw some insight on their best practices with a classic blog example.
+
+The idea is to show how similar they are and get you hyped enough to try it by yourself.
+
+### File structure
+
+![File structure comparison](images/phoenix-rails-project-structure.png)
+
+No opinionated framwework would be complete without a take on how files should be organized. Phoenix's structure inherits from how packages (Elixir's equivalent for gems) are structured, all custom codes goes inside the `lib/` folder. Inside it you'll find right away the main difference between both frameworks in this aspect: Phoenix separates the layer of web presentation from the business logic. In this example we have a `blog/` folder, which cointains the basic modules to build our business logic, and the `blog_web` which looks pretty similar to the Rails' `app/` directory. There are some subtle differences though but we'll take a deeper look into the difference in the following sections.
```

Comment:
Typo on the second line. Should be `all custom code` I think.

Reply:
When you say `In this example` in the middle of the paragraph, are you referring to the image placed above ?